encoder/vaapiencoder_vp9.cpp

```diff
@@ -0,0 +1,323 @@
+    m_videoParamCommon.rcParams.maxQP = kMaxQPValue;
+    m_videoParamCommon.rcParams.initQP = kDefaultQPValue;
+
+    //add extra surfaces to operate due to kMaxReferenceFrames
+    //vaapi_encoder class will create 5 extra surfaces already
+    m_maxOutputBuffer = kMaxReferenceFrames;
+}
+
+VaapiEncoderVP9::~VaapiEncoderVP9() {}
+
+YamiStatus VaapiEncoderVP9::getMaxOutSize(uint32_t* maxSize)
+{
+    FUNC_ENTER();
+    *maxSize = m_maxCodedbufSize;
+    return YAMI_SUCCESS;
+}
+
+void VaapiEncoderVP9::resetParams()
+{
+    m_maxCodedbufSize = width() * height() * 3 / 2;
+
+    // adding extra padding.  In particular small resolutions require more
+    // space depending on other quantization parameters during execution. The
+    // value below is a good compromise
+    m_maxCodedbufSize += 0x1000;
+}
+
+YamiStatus VaapiEncoderVP9::start()
+{
+    FUNC_ENTER();
+    resetParams();
+    return VaapiEncoderBase::start();
+}
+
+void VaapiEncoderVP9::flush()
+{
+    FUNC_ENTER();
+    m_frameCount = 0;
+    m_reference.clear();
+    VaapiEncoderBase::flush();
+}
+
+YamiStatus VaapiEncoderVP9::stop()
+{
+    flush();
+    return VaapiEncoderBase::stop();
+}
+
+YamiStatus VaapiEncoderVP9::setParameters(VideoParamConfigType type,
+                                          Yami_PTR videoEncParams)
+{
+    YamiStatus status = YAMI_SUCCESS;
+    FUNC_ENTER();
+    if (!videoEncParams)
+        return YAMI_INVALID_PARAM;
+
+    switch (type) {
+    default:
+        status = VaapiEncoderBase::setParameters(type, videoEncParams);
+        break;
+    }
+    return status;
+}
+
+YamiStatus VaapiEncoderVP9::getParameters(VideoParamConfigType type,
+                                          Yami_PTR videoEncParams)
+{
+    FUNC_ENTER();
+    if (!videoEncParams)
+        return YAMI_INVALID_PARAM;
+
+    return VaapiEncoderBase::getParameters(type, videoEncParams);
+}
+
+YamiStatus VaapiEncoderVP9::doEncode(const SurfacePtr& surface,
+                                     uint64_t timeStamp, bool forceKeyFrame)
+{
+    YamiStatus ret;
+    if (!surface)
+        return YAMI_INVALID_PARAM;
+
+    PicturePtr picture(new VaapiEncPictureVP9(m_context, surface, timeStamp));
+
+    m_frameCount %= keyFramePeriod();
+    picture->m_type = (m_frameCount ? VAAPI_PICTURE_P : VAAPI_PICTURE_I);
+    m_frameCount++;
+
+    CodedBufferPtr codedBuffer
+        = VaapiCodedBuffer::create(m_context, m_maxCodedbufSize);
+    if (!codedBuffer)
+        return YAMI_OUT_MEMORY;
+    picture->m_codedBuffer = codedBuffer;
+    codedBuffer->setFlag(ENCODE_BUFFERFLAG_ENDOFFRAME);
+    if (picture->m_type == VAAPI_PICTURE_I) {
+        codedBuffer->setFlag(ENCODE_BUFFERFLAG_SYNCFRAME);
+    }
+    ret = encodePicture(picture);
+    if (ret != YAMI_SUCCESS) {
+        return ret;
+    }
+    output(picture);
+    return YAMI_SUCCESS;
+}
+
+bool VaapiEncoderVP9::fill(VAEncSequenceParameterBufferVP9* seqParam) const
+{
+    seqParam->max_frame_width = kMaxWidth;
```

Comment:
From the comments in va_enc_vp9.h.  This field is for max resolution in video sequence. It has should not related to platform. For our case, we never change resolution in our encoder (at least in yami). So it should be width().